portality/static/doaj/js/available_facetviews/admin.journalarticle.facetview.js

```diff
@@ -1,248 +1,548 @@
+        if (!type || type.length === 0) {
+            return {
+                valid: false,
+                error_id: "any_type_error"
+            }
+        }
+        return {valid: true};
+    }
+
+    function typeSelected() {
+        var type = doaj.currentFVOptions.active_filters._type;
+        if (type && type.length > 0) {
+            return type[0];
+        }
+        return null;
+    }
+
+    var mfb = doaj.newMultiFormBox({
+        selector: "#admin-bulk-box",
+        widths: {
+            edit_metadata: "600px"
+        },
+        bindings : {
+            editor_group : function(context) {
+                autocomplete($('#editor_group', context), 'name', 'editor_group', 1, false);
+            },
+            edit_metadata : function(context) {
+                autocomplete($('#publisher', context), 'bibjson.publisher');
+                autocomplete($('#platform', context), 'bibjson.provider');
+                $('#country', context).select2();
+                autocomplete($('#owner', context), 'id', 'account');
+            }
+        },
+        validators : {
+            withdraw : journalSelected,
+            reinstate: journalSelected,
+            delete: anySelected,
+            note : function(context) {
+                var valid = journalSelected();
+                if (!valid.valid) {
+                    return valid;
+                }
+                var val = context.find("#note").val();
+                if (val === "") {
+                    return {valid: false};
+                }
+                return {valid: true};
+            },
+            editor_group : function(context) {
+                var valid = journalSelected();
+                if (!valid.valid) {
+                    return valid;
+                }
+                var val = context.find("#editor_group").val();
+                if (val === "") {
+                    return {valid: false};
+                }
+                return {valid: true};
+            },
+            edit_metadata : function(context) {
+                // first check that the journal has been selected
+                var valid = journalSelected();
+                if (!valid.valid) {
+                    return valid;
+                }
+
+                // now check that at least one field has been completed
+                var found = false;
+                var fields = ["#publisher", "#platform", "#country", "#owner", "#contact_name", "#contact_email", "#doaj_seal"];
+                for (var i = 0; i < fields.length; i++) {
+                    var val = context.find(fields[i]).val();
+                    if (val !== "") {
+                        found = true;
+                    }
+                }
+                if (!found) {
+                    return {valid: false};
+                }
+
+                // now check for valid field contents
+                // quick and dirty email check - this will be done properly server-side
```

Comment:
I would generally not attempt anything past name@domain.tld . See https://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address or many similar questions. It's best to not attempt to validate email via regex since you always exclude valid addresses, like `foo@emanuil@domain.com`